### DISCUSSION TITLE
Redis Backend Support

### DISCUSSION THREAD

**Discussion by jsmassa** [replikativ/datahike#432] [Ideas]
Should we support the Redis as backend for Datahike?

**Comment by whilo** [comment]
Redis provides faster durability at the cost of having to live in memory. I always thought that this trade-off would be particularly interesting for session stores (a use case where Redis is also traditionally used). Nonetheless you need to serialize and flush all changes in the hh-tree into the underlying Redis store on transactions and I think at the moment the IO costs don't dominate our throughput in most cases, it is therefore more important to first batch transactions in the transactor and work on the persistent set for higher read/write speed in general and then hit the IO costs for writing. But it should also not be a lot of work now to add Redis and if some user is interested I would cover it.
